### PR TITLE
denoiseprofile: preview more accurate

### PR DESCRIPTION
**Pull request by rawfiner**
Makes the denoiseprofile darkroom preview more accurate.
Still not perfect (I don't know if it is even possible), but better than before :-)

### CODE REVIEW

File: src/iop/denoiseprofile.c [hunk 1]
```diff
@@ -2518,9 +2522,9 @@ static int process_wavelets_cl(struct dt_iop_module_t *self, dt_dev_pixelpipe_io
     wb[2] = 2.0f * piece->pipe->dsc.processed_maximum[2];
   }
   // adaptive p depending on white balance
-  const float p[4] = { MAX(d->shadows - 0.1 * logf(wb[0]), 0.0f),
-                       MAX(d->shadows - 0.1 * logf(wb[1]), 0.0f),
-                       MAX(d->shadows - 0.1 * logf(wb[2]), 0.0f), 1.0f};
+  const float p[4] = { MAX(d->shadows - 0.1 * logf(wb[0]) + 0.1 * logf(scale), 0.0f),
```
Comment (TurboGit) [inline src/iop/denoiseprofile.c]: As above.

File: src/iop/denoiseprofile.c [hunk 2]
```diff
@@ -1728,11 +1729,12 @@ static void process_nlmeans_sse(struct dt_iop_module_t *self, dt_dev_pixelpipe_i
     for(int i = 0; i < 3; i++) wb[i] = piece->pipe->dsc.processed_maximum[i];
   }
   // adaptive p depending on white balance
-  const float p[3] = { MAX(d->shadows - 0.1 * logf(wb[0]), 0.0f),
-                       MAX(d->shadows - 0.1 * logf(wb[1]), 0.0f),
-                       MAX(d->shadows - 0.1 * logf(wb[2]), 0.0f)};
+  const float p[3] = { MAX(d->shadows - 0.1 * logf(wb[0]) + 0.1 * logf(scale), 0.0f) ,
```
Comment (TurboGit) [inline src/iop/denoiseprofile.c]: See below.

File: src/iop/denoiseprofile.c [hunk 3]
```diff
@@ -1505,12 +1505,13 @@ static void process_nlmeans(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t
...
```
Comment (TurboGit) [inline src/iop/denoiseprofile.c]: See below.

File: src/iop/denoiseprofile.c [hunk 4]
```diff
@@ -1301,9 +1301,9 @@ static void process_wavelets(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_
     wb[2] = 2.0f * piece->pipe->dsc.processed_maximum[2];
   }
   // adaptive p depending on white balance
-  const float p[3] = { MAX(d->shadows - 0.1 * logf(wb[0]), 0.0f),
-                       MAX(d->shadows - 0.1 * logf(wb[1]), 0.0f),
-                       MAX(d->shadows - 0.1 * logf(wb[2]), 0.0f)};
+  const float p[3] = { MAX(d->shadows - 0.1 * logf(wb[0]) + 0.1 * logf(in_scale), 0.0f) ,
```
Comment (TurboGit) [inline src/iop/denoiseprofile.c]: See below.

**Comment by rawfiner** [comment]
Thanks for the review, indeed the logs can be factorized using log properties. :-)
Done in this last commit.

**Comment by TurboGit** [comment]
Oh and you fixed my fix :) I didn't see it was minus as first operation :) Anyway, logf() is a quite CPU intensive computation better to have one removed. Thanks.

**Pull request merged**